### ISSUE TITLE
Upgrading from 0.1.23 to 0.2.3 does not show altair plots anymore.

### ISSUE THREAD

**Issue by afonit** [voila-dashboards/voila#722]
I posted this in the altair tracker and Jake suggested it may be an issue in voila so I am posting in here.
https://github.com/altair-viz/altair/issues/2298

Altair: 4.1.0
Voila: 0.2.3
jupyterlab: 2.2.8

I converted my setup to serve my dashboards with voila with **ssl** **https**, when I noticed some of the dashboards were missing their charts while others had the charts.

I finally figured out that if I showed the chart with an ipywidget then the chart would show up.
**I am asking for help/guidance in trying to figure out where the problem could be - for altair charts not showing up unless a widget is present when served through voila**

I will first show screenshots, then code and errors from the chrome dev tools console.

### First - what it looks like in jupyterlab:
![image](https://user-images.githubusercontent.com/1798497/94278851-53001100-ff19-11ea-9ff9-0b635583b807.png)

### Second - here is what it looks like in voila:
![image](https://user-images.githubusercontent.com/1798497/94278949-732fd000-ff19-11ea-8833-dfe7a72bbddc.png)

##### code for "Chart without Widget":
```python
import altair as alt
import pandas as pd

data = {'valuepre': [1, 2, 3, 4],
       'valuepost': [1,2,3,4]
       }

df = pd.DataFrame(data)

alt.Chart(df).mark_circle().encode(
    x='valuepre:Q',
    y='valuepost:Q')
```

##### code for "Chart with widget":
```python
import ipywidgets as widgets
from ipywidgets import interact
import altair as alt
import pandas as pd

data = {'valuepre': [1, 2, 3, 4],
       'valuepost': [1,2,3,4]
       }

df = pd.DataFrame(data)

def generate_chart(nothing):
    return alt.Chart(df).mark_circle().encode(
    x='valuepre:Q',
    y='valuepost:Q')

interact(generate_chart, nothing=widgets.Dropdown(value='b', options=['b']));
```

##### Console output on the voila page for "Chart without Widget":
```
voila_altair_no_widget.ipynb:297 Uncaught (in promise) ReferenceError: vegaEmbed is not defined
    at voila_altair_no_widget.ipynb:297
(anonymous) @ voila_altair_no_widget.ipynb:297
Promise.then (async)
(anonymous) @ voila_altair_no_widget.ipynb:297
(anonymous) @ voila_altair_no_widget.ipynb:299
DevTools failed to load SourceMap: Could not load content for https://cdn.jsdelivr.net/npm/vega.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
require.min.js:1 Uncaught Error: Mismatched anonymous define() module: function(e,t){"use strict";function r(e){if(e&&e.__esModule)return e;var t=Object.create(null);return e&&Object.keys(e).forEach((function(r){if("default"!==r){var n=Object.getOwnPropertyDescriptor(e,r);Object.defineProperty(t,r,n.get?n:{enumerable:!0,get:function(){return e[r]}})}})),t.default=e,Object.freeze(t)}var n=r(e),o=r(t),i=function(){return(i=Object.assign||function(e){for(var t,r=1,n=arguments.length;r<n;r++)for(var o in t=arguments[r])Object.prototype.hasOwnProperty.call(t,o)&&(e[o]=t[o]);return e}).apply(this,arguments)};function a(e,t,r,n){return new(r||(r=Promise))((function(o,i){function a(e){try{l(n.next(e))}catch(e){i(e)}}function s(e){try{l(n.throw(e))}catch(e){i(e)}}function l(e){var t;e.done?o(e.value):(t=e.value,t instanceof r?t:new r((function(e){e(t)}))).then(a,s)}l((n=n.apply(e,t||[])).next())}))}function s(e,t){var r,n,o,i,a={label:0,sent:function(){if(1&o[0])throw o[1];return o[1]},trys:[],ops:[]};return i={next:s(0),throw:s(1),return:s(2)},"function"==typeof Symbol&&(i[Symbol.iterator]=function(){return this}),i;function s(i){return function(s){return function(i){if(r)throw new TypeError("Generator is already executing.");for(;a;)try{if(r=1,n&&(o=2&i[0]?n.return:i[0]?n.throw||((o=n.return)&&o.call(n),0):n.next)&&!(o=o.call(n,i[1])).done)return o;switch(n=0,o&&(i=[2&i[0],o.value]),i[0]){case 0:case 1:o=i;break;case 4:return a.label++,{value:i[1],done:!1};case 5:a.label++,n=i[1],i=[0];continue;case 7:i=a.ops.pop(),a.trys.pop();continue;default:if(!(o=a.trys,(o=o.length>0&&o[o.length-1])||6!==i[0]&&2!==i[0])){a=0;continue}if(3===i[0]&&(!o||i[1]>o[0]&&i[1]<o[3])){a.label=i[1];break}if(6===i[0]&&a.label<o[1]){a.label=o[1],o=i;break}if(o&&a.label<o[2]){a.label=o[2],a.ops.push(i);break}o[2]&&a.ops.pop(),a.trys.pop();continue}i=t.call(e,a)}catch(e){i=[6,e],n=0}finally{r=o=0}if(5&i[0])throw i[1];return{value:i[0]?i[1]:void 0,done:!0}}([i,s])}}}
/*!
   * https://github.com/Starcounter-Jack/JSON-Patch
   * (c) 2017 Joachim Wester
   * MIT license
   */var l,c=(l=function(e,t){return(l=Object.setPrototypeOf||{__proto__:[]}instanceof Array&&function(e,t){e.__proto__=t}||function(e,t){for(var r in t)t.hasOwnProperty(r)&&(e[r]=t[r])})(e,t)},function(e,t){function r(){this.constructor=e}l(e,t),e.prototype=null===t?Object.create(t):(r.prototype=t.prototype,new r)}),p=Object.prototype.hasOwnProperty;function d(e,t){return p.call(e,t)}function u(e){if(Array.isArray(e)){for(var t=new Array(e.length),r=0;r<t.length;r++)t[r]=""+r;return t}if(Object.keys)return Object.keys(e);t=[];for(var n in e)d(e,n)&&t.push(n);return t}function f(e){switch(typeof e){case"object":return JSON.parse(JSON.stringify(e));case"undefined":return null;default:return e}}function h(e){for(var t,r=0,n=e.length;r<n;){if(!((t=e.charCodeAt(r))>=48&&t<=57))return!1;r++}return!0}function v(e){return-1===e.indexOf("/")&&-1===e.indexOf("~")?e:e.replace(/~/g,"~0").replace(/\//g,"~1")}function g(e){return e.replace(/~1/g,"/").replace(/~0/g,"~")}function m(e,t){var r=[e];for(var n in t){var o="object"==typeof t[n]?JSON.stringify(t[n],null,2):t[n];void 0!==o&&r.push(n+": "+o)}return r.join("\n")}var E=function(e){function t(t,r,n,o,i){var a=this.constructor,s=e.call(this,m(t,{name:r,index:n,operation:o,tree:i}))||this;return s.name=r,s.index=n,s.operation=o,s.tree=i,Object.setPrototypeOf(s,a.prototype),s.message=m(t,{name:r,index:n,operation:o,tree:i}),s}return c(t,e),t}(Error),b=E,y=f,w={add:function(e,t,r){return e[t]=this.value,{newDocument:r}},remove:function(e,t,r){var n=e[t];return delete e[t],{newDocument:r,removed:n}},replace:function(e,t,r){var n=e[t];return e[t]=this.value,{newDocument:r,removed:n}},move:function(e,t,r){var n=I(r,this.path);n&&(n=f(n));var o=A(r,{op:"remove",path:this.from}).removed;return A(r,{op:"add",path:this.path,value:o}),{newDocument:r,removed:n}},copy:function(e,t,r){var n=I(r,this.from);return A(r,{op:"add",path:this.path,value:f(n)}),{newDocument:r}},test:function(e,t,r){return{newDocument:r,test:L(e[t],this.value)}},_get:function(e,t,r){return this.value=e[t],{newDocument:r}}},O={add:function(e,t,r){return h(t)?e.splice(t,0,this.value):e[t]=this.value,{newDocument:r,index:t}},remove:function(e,t,r){return{newDocument:r,removed:e.splice(t,1)[0]}},replace:function(e,t,r){var n=e[t];return e[t]=this.value,{newDocument:r,removed:n}},move:w.move,copy:w.copy,test:w.test,_get:w._get};function I(e,t){if(""==t)return e;var r={op:"_get",path:t};return A(e,r),r.value}function A(e,t,r,n,o,i){if(void 0===r&&(r=!1),void 0===n&&(n=!0),void 0===o&&(o=!0),void 0===i&&(i=0),r&&("function"==typeof r?r(t,0,e,t.path):N(t,0)),""===t.path){var a={newDocument:e};if("add"===t.op)return a.newDocument=t.value,a;if("replace"===t.op)return a.newDocument=t.value,a.removed=e,a;if("move"===t.op||"copy"===t.op)return a.newDocument=I(e,t.from),"move"===t.op&&(a.removed=e),a;if("test"===t.op){if(a.test=L(e,t.value),!1===a.test)throw new b("Test operation failed","TEST_OPERATION_FAILED",i,t,e)require.min.js:1)
    at v (require.min.js:1)
    at Object.o [as require] (require.min.js:1)
    at requirejs (require.min.js:1)
    at main.js?v=9de22d21d90835225053df83a51fd350:10
makeError @ require.min.js:1
v @ require.min.js:1
o @ require.min.js:1
requirejs @ require.min.js:1
(anonymous) @ main.js?v=9de22d21d90835225053df83a51fd350:10
DevTools failed to load SourceMap: Could not load content for https://cdn.jsdelivr.net/npm/vega-embed.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load SourceMap: Could not load content for chrome-extension://ndjpnladcallmjemlbaebfadecfhkepb/editor/config.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
DevTools failed to load SourceMap: Could not load content for chrome-extension://ndjpnladcallmjemlbaebfadecfhkepb/editor/content.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
favicon.ico:1 GET https://voila.comcast.com/favicon.ico 404 (Not Found)
```

##### Console output for "Chart with widget":
```
Starting WebSocket: wss://voila.comcast.com/api/kernels/afd66be3-bbde-4305-86c8-0d2b1b6b4038
default.js:1452 Kernel: connected (afd66be3-bbde-4305-86c8-0d2b1b6b4038)
DevTools failed to load SourceMap: Could not load content for chrome-extension://ndjpnladcallmjemlbaebfadecfhkepb/editor/config.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
DevTools failed to load SourceMap: Could not load content for chrome-extension://ndjpnladcallmjemlbaebfadecfhkepb/editor/content.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
DevTools failed to load SourceMap: Could not load content for https://cdn.jsdelivr.net/npm/vega-embed.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load SourceMap: Could not load content for https://cdn.jsdelivr.net/npm/vega.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
```

_It is unclear to me why altair charts are no longer showing up unless they are wrapped in an ipywidget._

**Comment by afonit** [comment]
This is only when serving over ssl
if I serve over http then the chart shows up fine **without** the ipywidget.

but If I serve voila over https
then that is when I get the above errors unless I embed the plot in an ipywidget.

the ssl part for /etc/nginx/sites-available/voila is:
```
server {
    listen 443 ssl;
    server_name voila.comcast.com;

    ssl_protocols TLSv1.2;
    ssl_prefer_server_ciphers on;
    ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';

    ssl_certificate /official_cert_signed/voila.comcast.com.crt;
    ssl_certificate_key /official_cert_signed/voila.comcast.com.key;

    proxy_buffering off;
...
```

Any ideas on why those errors would be there preventing voila from serving the chart without it being embedded in a widget?

**Comment by afonit** [comment]
Another odd thing,
for the no-widget example,:
* If there is only an altair plot without a widget, the plot does not show up
* if I put a **bqplot** chart in the notebook along with the altair chart, then the altair chart also shows up.

![image](https://user-images.githubusercontent.com/1798497/94371692-55539e00-00c6-11eb-867a-6b5a2e525429.png)

![image](https://user-images.githubusercontent.com/1798497/94371713-83d17900-00c6-11eb-88d6-e0f1c93eb590.png)


But if I remove the bqplot example from the notebook and just have the altair chart, then the altair plot does **not** show up as demonstrated in the original post.

**Comment by afonit** [comment]
I will close this until I can get more concrete details since this seems to be inconsistent right now.

**Comment by afonit** [comment]
Does voila do anything to the urls?
Here is a comment from one of the vega developers saying that:
https://github.com/vega/vega-embed/issues/557#issuecomment-700053232

![image](https://user-images.githubusercontent.com/1798497/94449441-12eb9900-017a-11eb-8b7c-181a25880f25.png)

**Comment by afonit** [comment]
I realized I had changed quite a few things at once, and I needed to start from scratch and see what item specifically may be contributing to the issue.

The first thing I tried, looks like it is it.

I was on voila **0.1.23**

So I have that version on my local instance, and the plots show up fine with voila.

Then I **upgrade voila to 0.2.3** and the plots do not show up:
```
DevTools failed to load SourceMap: Could not load content for chrome-extension://ndjpnladcallmjemlbaebfadecfhkepb/editor/config.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
DevTools failed to load SourceMap: Could not load content for chrome-extension://ndjpnladcallmjemlbaebfadecfhkepb/editor/content.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
DevTools failed to load SourceMap: Could not load content for https://cdn.jsdelivr.net/npm/vega.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
x_voila_altair_text.ipynb:290 Uncaught (in promise) ReferenceError: vegaEmbed is not defined
    at x_voila_altair_text.ipynb:290
(anonymous) @ x_voila_altair_text.ipynb:290
Promise.then (async)
(anonymous) @ x_voila_altair_text.ipynb:290
(anonymous) @ x_voila_altair_text.ipynb:292
DevTools failed to load SourceMap: Could not load content for https://cdn.jsdelivr.net/npm/vega-embed.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
main.js?v=9de22d21d90835225053df83a51fd350:13 Uncaught (in promise) TypeError: voila.connectKernel is not a function
    at main.js?v=9de22d21d90835225053df83a51fd350:13
    at main.js?v=9de22d21d90835225053df83a51fd350:55
    at Object.execCb (require.min.js:1)
    at b.check (require.min.js:1)
    at b.<anonymous> (require.min.js:1)
    at require.min.js:1
    at require.min.js:1
    at each (require.min.js:1)
    at b.emit (require.min.js:1)
    at b.check (require.min.js:1)
(anonymous) @ main.js?v=9de22d21d90835225053df83a51fd350:13
(anonymous) @ main.js?v=9de22d21d90835225053df83a51fd350:55
execCb @ require.min.js:1
check @ require.min.js:1
(anonymous) @ require.min.js:1
(anonymous) @ require.min.js:1
(anonymous) @ require.min.js:1
each @ require.min.js:1
emit @ require.min.js:1
check @ require.min.js:1
enable @ require.min.js:1
init @ require.min.js:1
h @ require.min.js:1
completeLoad @ require.min.js:1
onScriptLoad @ require.min.js:1
load (async)
req.load @ require.min.js:1
load @ require.min.js:1
load @ require.min.js:1
fetch @ require.min.js:1
check @ require.min.js:1
enable @ require.min.js:1
enable @ require.min.js:1
(anonymous) @ require.min.js:1
(anonymous) @ require.min.js:1
each @ require.min.js:1
enable @ require.min.js:1
init @ require.min.js:1
(anonymous) @ require.min.js:1
setTimeout (async)
req.nextTick @ require.min.js:1
o @ require.min.js:1
requirejs @ require.min.js:1
(anonymous) @ main.js?v=9de22d21d90835225053df83a51fd350:10
favicon.ico:1 GET http://localhost:8866/favicon.ico 404 (Not Found)
```

**Comment by afonit** [comment]
I thought it would be a good idea to see if I could nail this down to occurring between two versions.

With a clean environment I installed:
voila 0.1.23

Altair plots showed up fine.

I then installed:
voila 0.2.0

Altair plots do not show up,
here is the message from 0.2.0:
```
theme-light.css:1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)
2index.css:1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)
theme-light.css:1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)
require.min.js:1 GET http://localhost:8866/voila/templates/lab/static/main.js?v=9de22d21d90835225053df83a51fd350 net::ERR_ABORTED 500 (Internal Server Error)
req.load @ require.min.js:1
load @ require.min.js:1
load @ require.min.js:1
fetch @ require.min.js:1
check @ require.min.js:1
enable @ require.min.js:1
enable @ require.min.js:1
(anonymous) @ require.min.js:1
(anonymous) @ require.min.js:1
each @ require.min.js:1
enable @ require.min.js:1
init @ require.min.js:1
(anonymous) @ require.min.js:1
setTimeout (async)
req.nextTick @ require.min.js:1
o @ require.min.js:1
requirejs @ require.min.js:1
(anonymous) @ x_voila_altair_text.ipynb:325
require.min.js:1 Uncaught Error: Script error for "/voila/templates/lab/static/main.js?v=9de22d21d90835225053df83a51fd350"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.min.js:1)
    at HTMLScriptElement.onScriptError (require.min.js:1)
makeError @ require.min.js:1
onScriptError @ require.min.js:1
error (async)
req.load @ require.min.js:1
load @ require.min.js:1
load @ require.min.js:1
fetch @ require.min.js:1
check @ require.min.js:1
enable @ require.min.js:1
enable @ require.min.js:1
(anonymous) @ require.min.js:1
(anonymous) @ require.min.js:1
each @ require.min.js:1
enable @ require.min.js:1
init @ require.min.js:1
(anonymous) @ require.min.js:1
setTimeout (async)
req.nextTick @ require.min.js:1
o @ require.min.js:1
requirejs @ require.min.js:1
(anonymous) @ x_voila_altair_text.ipynb:325
x_voila_altair_text.ipynb:289 Uncaught (in promise) ReferenceError: vegaEmbed is not defined
    at x_voila_altair_text.ipynb:289
(anonymous) @ x_voila_altair_text.ipynb:289
Promise.then (async)
(anonymous) @ x_voila_altair_text.ipynb:289
(anonymous) @ x_voila_altair_text.ipynb:291
DevTools failed to load SourceMap: Could not load content for https://cdn.jsdelivr.net/npm/vega.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load SourceMap: Could not load content for chrome-extension://ndjpnladcallmjemlbaebfadecfhkepb/editor/config.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
DevTools failed to load SourceMap: Could not load content for chrome-extension://ndjpnladcallmjemlbaebfadecfhkepb/editor/content.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
DevTools failed to load SourceMap: Could not load content for https://cdn.jsdelivr.net/npm/vega-embed.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
```

So it appears something was changed between 0.1.23 and 0.2.0 that broke the functionality which allowed altair plots to be served and show up.

**Comment by afonit** [comment]
I have got this down to this reproducible example:

First I setup my environment:
```
conda create -n voila_altair_test python=3.7
conda activate voila_altair_test
conda install -c conda-forge altair
pip install voila==0.1.23
```

I then create a single notebook file with a single cell that has the below content:
```
import altair as alt
import pandas as pd

data = {'valuepre': [1, 2, 3, 4],
       'valuepost': [1,2,3,4]
       }

df = pd.DataFrame(data)

alt.Chart(df).mark_circle().encode(
    x='valuepre:Q',
    y='valuepost:Q')
```

I then launch voila from the terminal:
```
voila
```

After navigating to the notebook I created - the plot shows up fine:
![image](https://user-images.githubusercontent.com/1798497/94558839-5f90ac00-022e-11eb-8292-d97834918b0a.png)


I then cancel the running voila instance and install 0.2.0
```
pip install voila==0.2.0
```

now I run voila again:
```
voila
```
After navigating to the notebook I do not get the chart:
![image](https://user-images.githubusercontent.com/1798497/94559018-9ff02a00-022e-11eb-8dce-3f497b7a5c1c.png)

Here is the chrome dev console output:
```
theme-light.css:1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)
index.css:1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)
main.js:1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)
require.min.js:1 Uncaught Error: Script error for "/voila/templates/lab/static/main.js?v=9de22d21d90835225053df83a51fd350"
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.min.js:1)
    at HTMLScriptElement.onScriptError (require.min.js:1)
x_voila_altair_text.ipynb:289 Uncaught (in promise) ReferenceError: vegaEmbed is not defined
    at x_voila_altair_text.ipynb:289
index.css:1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)
theme-light.css:1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)
DevTools failed to load SourceMap: Could not load content for chrome-extension://ndjpnladcallmjemlbaebfadecfhkepb/editor/config.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
DevTools failed to load SourceMap: Could not load content for chrome-extension://ndjpnladcallmjemlbaebfadecfhkepb/editor/content.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
DevTools failed to load SourceMap: Could not load content for https://cdn.jsdelivr.net/npm/vega.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
DevTools failed to load SourceMap: Could not load content for https://cdn.jsdelivr.net/npm/vega-embed.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
```

And here is the terminal output while running voila and loading that notebook:
```
[Voila] Voilà is running at:
http://localhost:8866/
WARNING:tornado.general:404 GET /api/kernels/62cd20a6-6663-461e-85b7-522030401dd1/channels?session_id=8727b4c1-0a7d-4975-91ed-7e3c5c21754c (127.0.0.1): Kernel does not exist: 62cd20a6-6663-461e-85b7-522030401dd1
WARNING:tornado.access:404 GET /api/kernels/62cd20a6-6663-461e-85b7-522030401dd1/channels?session_id=8727b4c1-0a7d-4975-91ed-7e3c5c21754c (127.0.0.1) 37.91ms
[Voila] Kernel started: f13e6bbb-a644-4a19-b154-82334a795b96
ERROR:tornado.application:Uncaught exception GET /voila/templates/lab/static/theme-light.css?v=25cb4a7071f905b9bdcf234d7215aa07 (::1)
HTTPServerRequest(protocol='http', host='localhost:8866', method='GET', uri='/voila/templates/lab/static/theme-light.css?v=25cb4a7071f905b9bdcf234d7215aa07', version='HTTP/1.1', remote_ip='::1')
Traceback (most recent call last):
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\tornado\web.py", line 1703, in _execute
    result = await result
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\tornado\web.py", line 2580, in get
    absolute_path = self.get_absolute_path(self.root, self.path)
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\voila\static_file_handler.py", line 56, in get_absolute_path
    template, static, relpath = path.split('/', 2)
ValueError: not enough values to unpack (expected 3, got 1)
ERROR:tornado.access:500 GET /voila/templates/lab/static/theme-light.css?v=25cb4a7071f905b9bdcf234d7215aa07 (::1) 18.81ms
ERROR:tornado.application:Uncaught exception GET /voila/templates/lab/static/index.css?v=157743fa4e0f4c8216448c596a8d3586 (::1)
HTTPServerRequest(protocol='http', host='localhost:8866', method='GET', uri='/voila/templates/lab/static/index.css?v=157743fa4e0f4c8216448c596a8d3586', version='HTTP/1.1', remote_ip='::1')
Traceback (most recent call last):
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\tornado\web.py", line 1703, in _execute
    result = await result
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\tornado\web.py", line 2580, in get
    absolute_path = self.get_absolute_path(self.root, self.path)
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\voila\static_file_handler.py", line 56, in get_absolute_path
    template, static, relpath = path.split('/', 2)
ValueError: not enough values to unpack (expected 3, got 1)
ERROR:tornado.access:500 GET /voila/templates/lab/static/index.css?v=157743fa4e0f4c8216448c596a8d3586 (::1) 31.91ms
ERROR:tornado.application:Uncaught exception GET /voila/templates/lab/static/main.js?v=9de22d21d90835225053df83a51fd350 (::1)
HTTPServerRequest(protocol='http', host='localhost:8866', method='GET', uri='/voila/templates/lab/static/main.js?v=9de22d21d90835225053df83a51fd350', version='HTTP/1.1', remote_ip='::1')
Traceback (most recent call last):
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\tornado\web.py", line 1703, in _execute
    result = await result
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\tornado\web.py", line 2580, in get
    absolute_path = self.get_absolute_path(self.root, self.path)
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\voila\static_file_handler.py", line 56, in get_absolute_path
    template, static, relpath = path.split('/', 2)
ValueError: not enough values to unpack (expected 3, got 1)
ERROR:tornado.access:500 GET /voila/templates/lab/static/main.js?v=9de22d21d90835225053df83a51fd350 (::1) 6.99ms
[Voila] WARNING | Replacing stale connection: 62cd20a6-6663-461e-85b7-522030401dd1:8727b4c1-0a7d-4975-91ed-7e3c5c21754c
ERROR:tornado.application:Uncaught exception GET /voila/templates/lab/static/theme-light.css?v=25cb4a7071f905b9bdcf234d7215aa07 (::1)
HTTPServerRequest(protocol='http', host='localhost:8866', method='GET', uri='/voila/templates/lab/static/theme-light.css?v=25cb4a7071f905b9bdcf234d7215aa07', version='HTTP/1.1', remote_ip='::1')
Traceback (most recent call last):
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\tornado\web.py", line 1703, in _execute
    result = await result
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\tornado\web.py", line 2580, in get
    absolute_path = self.get_absolute_path(self.root, self.path)
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\voila\static_file_handler.py", line 56, in get_absolute_path
    template, static, relpath = path.split('/', 2)
ValueError: not enough values to unpack (expected 3, got 1)
ERROR:tornado.access:500 GET /voila/templates/lab/static/theme-light.css?v=25cb4a7071f905b9bdcf234d7215aa07 (::1) 7.89ms
ERROR:tornado.application:Uncaught exception GET /voila/templates/lab/static/index.css?v=157743fa4e0f4c8216448c596a8d3586 (::1)
HTTPServerRequest(protocol='http', host='localhost:8866', method='GET', uri='/voila/templates/lab/static/index.css?v=157743fa4e0f4c8216448c596a8d3586', version='HTTP/1.1', remote_ip='::1')
Traceback (most recent call last):
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\tornado\web.py", line 1703, in _execute
    result = await result
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\tornado\web.py", line 2580, in get
    absolute_path = self.get_absolute_path(self.root, self.path)
  File "c:\users\jthomp261\appdata\local\continuum\anaconda3\envs\voila_altair_test\lib\site-packages\voila\static_file_handler.py", line 56, in get_absolute_path
    template, static, relpath = path.split('/', 2)
ValueError: not enough values to unpack (expected 3, got 1)
ERROR:tornado.access:500 GET /voila/templates/lab/static/index.css?v=157743fa4e0f4c8216448c596a8d3586 (::1) 5.98ms
[Voila] WARNING | Replacing stale connection: 62cd20a6-6663-461e-85b7-522030401dd1:8727b4c1-0a7d-4975-91ed-7e3c5c21754c
```


Note: I can then do,
```
pip install voila==0.1.23
```
to downgrade back to voila 0.1.23 from 2.0 and I get the plot again.

So, something has changed from version **0.1.23** to version **2.0** that does not allow voila to display altair charts.

**Comment by SylvainCorlay** [comment]
@afonit this was an issue with tornado 6, and should be fixed in Voilà 0.2.3.

I just checked locally and seems to be working fine.

Let me know how it goes!

**Comment by afonit** [comment]
Thank you sylvian, in my clean environment that does work now.  So in my regular environment something must have gotten messed up with the upgrade to 0.2.3.  I will rebuild it.
Thank you.
